src/main/java/org/openstreetmap/atlas/generator/tools/caching/HadoopAtlasFileCache.java

```diff
@@ -55,9 +61,14 @@ public HadoopAtlasFileCache(final String parentAtlasPath,
      */
     public Optional<Resource> get(final String country, final Shard shard)
     {
+        String compiledAtlasScheme = "";
+        if (shard instanceof SlippyTile)
+        {
+            compiledAtlasScheme = this.atlasScheme.compile((SlippyTile) shard);
+        }
         final String atlasName = String.format("%s_%s", country, shard.getName());
-        final String atlasURIString = this.parentAtlasPath + "/" + country + "/" + atlasName
-                + FileSuffix.ATLAS.toString();
+        final String atlasURIString = this.parentAtlasPath + "/" + country + "/"
+                + compiledAtlasScheme + atlasName + FileSuffix.ATLAS.toString();
```

Comment:
My plan for the next update was to change all the `+ "/" +` constructs to `SparkFileHelper.combine()` calls. Since it looks like `compiledAtlasScheme` already has a trailing `/`, we'll have to figure out what to do here!

Reply:
Maybe skip the `SparkFileHelper.combine()`?

Reply:
I suppose so. I just always tend to recoil from hardcoding path separators. What if somebody wants to use Windows ;)